src/mlpack/methods/ann/layer/max_pooling.hpp

```diff
@@ -70,6 +70,26 @@ class MaxPooling
              const size_t strideHeight = 1,
              const bool floor = true);
 
+  /**
+   * Create the MaxPooling object for adaptive pooling.
+   *
+   * @param outputWidth Width of the output.
+   * @param outputHeight Height of the output.
+   * @param adaptivePooling Needs to be set to true to use adaptive pooling.
+   */
+  MaxPooling(const size_t outputWidth,
+             const size_t outputHeight,
+             const bool  adaptivePooling);
```

Comment:
IMO this should not be done since in `MaxPooling(10, 10)` the 10 would refer to the kernel width and height and if we use `MaxPooling(10,10, true)` here the 10 would refer to the output width and height which would be very confusing for the user.

Reply:
Hi @saksham189, what do you suggest?
1. A wrapper (A lot easier for user to understand, will end up containing a bit more code).
2. Switch to a string instead of a boolean. (That doesn't make much sense either but I guess it would have the least amount of redundant code.)
I would really appreciate your opinion. Thanks.

Reply:
Hi @walragatver, could you share your opinion on this. Thanks a lot.

Reply:
I guess it makes sense to have a separate layer for adaptive pooling as a wrapper over pooling layer.

Reply:
Great I will make the changes by tonight then. Thanks a ton.

Reply:
Hi @saksham189 , @birm , I have made the changes. Kindly have a look when you get the chance. Thanks a lot.